helix-term/src/commands.rs

```diff
@@ -3433,7 +3437,11 @@ fn paste_impl(
         .iter()
         .any(|value| get_line_ending_of_str(value).is_some());
 
-    let mut values = values.iter().cloned().map(Tendril::from).chain(repeat);
+    let mut values = values
+        .iter()
+        .map(|value| PASTE_LINE_ENDING_REGEX.replace_all(value, doc.line_ending.as_str()))
```

Comment:
In the future, this could probably be optimized by running `map_while` when `linewise` is true.